server/src/main/java/org/opensearch/rest/pagination/PaginatedQueryResponse.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.pagination;
+
+/**
+ * Pagination response metadata for a paginated query.
+ * @opensearch.internal
+ */
+public class PaginatedQueryResponse {
```

Comment:
The class name is little confusing. This is essentially giving nextToken only. Shouls this be PaginationNextToken